src/Nethermind/Nethermind.Evm/BlobGasCalculator.cs

```diff
@@ -31,12 +31,11 @@ public static ulong CalculateBlobGas(Transaction[] transactions)
 
     public static bool TryCalculateBlobGasPrice(BlockHeader header, Transaction transaction, out UInt256 blobGasPrice)
     {
-        if (!TryCalculateBlobGasPricePerUnit(header.ExcessBlobGas.Value, out UInt256 blobGasPricePerUnit))
-        {
-            blobGasPrice = UInt256.MaxValue;
-            return false;
-        }
-        return !UInt256.MultiplyOverflow(CalculateBlobGas(transaction), blobGasPricePerUnit, out blobGasPrice);
+        if (header.ExcessBlobGas != null &&
```

Comment:
I think this shouldn't be called if it is null in the first place

Reply:
So if any of transactions in block support blobs, you need to set ExcessBlobGas for the BlockHeader

Reply:
Or better - add it if spec.IsEip4844Enabled

Reply:
Yep - restored, fixed